src/addr.rs

```diff
@@ -322,6 +326,44 @@ impl Sub<VirtAddr> for VirtAddr {
     }
 }
 
+#[cfg(feature = "step_trait")]
+impl Step for VirtAddr {
+    fn steps_between(start: &Self, end: &Self) -> Option<usize> {
+        let mut steps = end.0.checked_sub(start.0)?;
+
+        // Check if we jumped the gap.
+        if end.0.get_bit(47) && !start.0.get_bit(47) {
+            steps -= 0xffff_0000_0000_0000;
+        }
+
+        usize::try_from(steps).ok()
+    }
+
+    fn forward_checked(start: Self, count: usize) -> Option<Self> {
+        let offset = u64::try_from(count).ok()?;
+        let mut addr = start.0.checked_add(offset)?;
+
+        // Jump the gap by sign extending the 47th bit.
+        if addr.get_bits(47..) == 0x1 {
+            addr.set_bits(47.., 0x1ffff);
+        }
+
+        Some(Self(addr))
+    }
+
+    fn backward_checked(start: Self, count: usize) -> Option<Self> {
+        let offset = u64::try_from(count).ok()?;
+        let mut addr = start.0.checked_sub(offset)?;
+
+        // Jump the gap by sign extending the 47th bit.
+        if addr.get_bits(47..) == 0x1fffe {
```

Comment:
Same concern as above: If count is too large, this get_bits might return something smaller than 0x1fffe.